src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -483,6 +483,108 @@ public static void Options_JsonSerializerContext_DoesNotFallbackToReflection()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(unsupportedValue, options));
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public static void Options_DisablingUseReflectionDefaultSwitch_DefaultOptionsDoesNotSupportReflection()
+        {
+            var options = new RemoteInvokeOptions
+            {
+                RuntimeConfigurationOptions =
+                {
+                    ["System.Text.Json.Serialization.UseReflectionDefault"] = false
+                }
+            };
+
+            RemoteExecutor.Invoke(static () =>
+            {
+                var options = JsonSerializerOptions.Default;
+                Assert.True(options.IsReadOnly);
+
+                Assert.NotNull(options.TypeInfoResolver);
+                Assert.True(options.TypeInfoResolver is not DefaultJsonTypeInfoResolver);
+                IList<IJsonTypeInfoResolver> resolverList = Assert.IsAssignableFrom<IList<IJsonTypeInfoResolver>>(options.TypeInfoResolver);
+
+                Assert.Empty(resolverList);
+                Assert.Empty(options.TypeInfoResolverChain);
+
+                Assert.Throws<NotSupportedException>(() => options.GetTypeInfo(typeof(string)));
+                Assert.Throws<NotSupportedException>(() => options.GetConverter(typeof(string)));
+
+                Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize("string"));
+                Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize("string", options));
+
+                Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<string>("\"string\""));
+                Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<string>("\"string\"", options));
+
+            }, options).Dispose();
+        }
+
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public static void Options_DisablingUseReflectionDefaultSwitch_NewOptionsDoesNotSupportReflection()
```

Comment:
What is the expected behavior when the switch is `true`?

Reply:
Do you mean, this call `options.GetTypeInfo(typeof(string)` won't throw since it will use reflection?

Reply:
Unfortunately, no. The original proposal in https://github.com/dotnet/runtime/issues/83279 attempted to do this but we had to scale back the ambition because it created problems with how the default `JsonSerializerOptions` constructor behaved in object initializer expressions. Once https://github.com/dotnet/runtime/issues/81269 is implemented the method will be returning `null` instead of throwing though.